### ISSUE TITLE
Create new, init, and free functions for edges and tests for these functions

### ISSUE THREAD

**Issue by williamhu314** [uchicago-cs/chiventure#1446]
We have decided to make the edges field within the specgraph struct a dynamically allocated 2D array. However, this make it more difficult for the user to create the adjacency matrix. In this issue, we aim to create new, init, and free functions for the edges adjacency matrix within the specgraph struct to make it easier to create the adjacency matrix. We also aim to create tests for each of these functions.

**Comment by rohandeme** [comment]
Created a new branch called autogenerate/edges to work on this issue.

**Comment by williamhu314** [comment]
We finished the functions and tests for init and new and are now working on the free function and test

**Comment by williamhu314** [comment]
The new function will allow the game author to pass a heap-allocated 1D array and the number of rows and columns they want in the 2D array and will output a heap-allocated 2D array. This will make things less complex because the game author will only need to use malloc once instead of implementing a for loop of mallocs in the case of a heap-allocated 2D array.

**Comment by williamhu314** [comment]
The free function will also allow the game author to avoid using a for loop when freeing an edges array.

**Comment by rohandeme** [comment]
We have finished writing the tests for free and init, thus concluding this issue. While the  attribute "edges" doesn't has its own struct, these functions will make it easier on the game author as they will only have to input a 1D array of "edges", which will then be converted into a 2D adjacency matrix through the functions we just wrote. A pull request will soon be made to transfer these changes into the "dev" branch, and in the future we may aim to add more tests for these three functions.
